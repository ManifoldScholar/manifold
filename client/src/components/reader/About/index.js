import Overlay from "./Overlay";
import Detail from "./Detail";
import Meta from "./Meta";

export default {
  Overlay,
  Detail,
  Meta
};
