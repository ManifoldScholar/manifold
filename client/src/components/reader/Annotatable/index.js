import Annotatable from './Annotatable';
export default Annotatable;

