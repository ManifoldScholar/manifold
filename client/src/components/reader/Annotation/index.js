import Comment from './Comment';
import Popup from './Popup';
import Selection from './Selection';

export default {
  Comment,
  Popup,
  Selection
};
