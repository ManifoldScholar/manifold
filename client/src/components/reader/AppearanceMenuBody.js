import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';

export default class AppearanceMenuBody extends Component {

  static propTypes = {
    appearance: PropTypes.object,
    selectFont: PropTypes.func,
    incrementFontSize: PropTypes.func,
    decrementFontSize: PropTypes.func,
    setColorScheme: PropTypes.func
  };

  setColorHandler = (event, scheme) => {
    event.stopPropagation();
    this.props.setColorScheme(scheme);
  };

  selectFontHandler = (event, font) => {
    event.stopPropagation();
    this.props.selectFont(font);
  };

  incrementSizeHandler = (event) => {
    event.stopPropagation();
    this.props.incrementFontSize();
  };

  decrementSizeHandler = (event) => {
    event.stopPropagation();
    this.props.decrementFontSize();
  };

  handleSerifButtonClick = (event) => {
    this.selectFontHandler(event, 'serif');
  };

  handleSansSerifButtonClick = (event) => {
    this.selectFontHandler(event, 'sans-serif');
  };

  handleLightButtonClick = (event) => {
    this.setColorHandler(event, 'light');
  };

  handleDarkButtonClick = (event) => {
    this.setColorHandler(event, 'dark');
  };


  render = () => {
    const typography = this.props.appearance.typography;
    const colorScheme = this.props.appearance.colors.colorScheme;

    // Conditional Classes
    const serifButtonClass = classNames({
      'font-type serif': true,
      'button-active': typography.font === 'serif'
    });
    const sansSerifButtonClass = classNames({
      'font-type sans-serif': true,
      'button-active': typography.font === 'sans-serif'
    });
    const lightSchemeButtonClass = classNames({
      light: true,
      'button-active': colorScheme === 'light'
    });
    const darkSchemeButtonClass = classNames({
      dark: true,
      'button-active': colorScheme === 'dark'
    });

    return (
        <nav className="appearance-menu">
          <div className="control-font">
            <button
              className={serifButtonClass}
              onClick={this.handleSerifButtonClick}
            >
              {'Serif'}
            </button>

            <div className="font-size">
              <button
                disabled={typography.size === typography.sizeMin ||
                  typography.font === 'sans-serif'}
                onClick={this.decrementSizeHandler}
              >
                <i className="manicon manicon-dash"></i>
                <span className="screen-reader-text">
                  {'Click to decrease font-size'}
                </span>
              </button>
              <button
                disabled={typography.size === typography.sizeMax ||
                  typography.font === 'sans-serif'}
                onClick={this.incrementSizeHandler}
              >
                <i className="manicon manicon-plus"></i>
                <span className="screen-reader-text">
                  {'Click to increase font-size'}
                </span>
              </button>
            </div>
          </div>
          <div className="control-font">
            <button className={sansSerifButtonClass}
              onClick={this.handleSansSerifButtonClick}
            >
              {'Sans Serif'}
            </button>

            <div className="font-size">
              <button
                disabled={typography.size === typography.sizeMin ||
                  typography.font === 'serif'}
                onClick={this.decrementSizeHandler}
              >
                <i className="manicon manicon-dash"></i>
                <span className="screen-reader-text">
                  {'Click to decrease font-size'}
                </span>
              </button>
              <button
                disabled={typography.size === typography.sizeMax ||
                  typography.font === 'serif'}
                onClick={this.incrementSizeHandler}
              >
                <i className="manicon manicon-plus"></i>
                <span className="screen-reader-text">
                  {'Click to increase font-size'}
                </span>
              </button>
            </div>
          </div>

          <div className="control-color">
            <button className={lightSchemeButtonClass}
              onClick={this.handleLightButtonClick}
            >
              <i className="manicon manicon-check"></i>
              <span className="screen-reader-text">
                {'Click to use light color scheme in reader'}
              </span>
            </button>
            <button className={darkSchemeButtonClass}
              onClick={this.handleDarkButtonClick}
            >
              <i className="manicon manicon-check"></i>
              <span className="screen-reader-text">
                {'Click to use dark color scheme in reader'}
              </span>
            </button>
          </div>
        </nav>
    );
  };
}
