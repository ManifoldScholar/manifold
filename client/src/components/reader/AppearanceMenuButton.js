import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';

export default class AppearanceMenuButton extends Component {

  static propTypes = {
    toggleAppearanceMenu: PropTypes.func,
    active: PropTypes.bool
  };

  clickHandler = (event) => {
    event.stopPropagation();
    this.props.toggleAppearanceMenu();
  };

  render() {
    const buttonClass = classNames({
      'button-appearance': true,
      'button-active': this.props.active
    });
    return (
        <button className={buttonClass} onClick={this.clickHandler}>
          <i className="manicon manicon-aa"></i>
          <span className="screen-reader-text">{'Click to open reader appearance menu'}</span>
        </button>
    );
  }
}
