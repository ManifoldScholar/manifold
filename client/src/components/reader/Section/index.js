import BodyNodes from './BodyNodes';
import Body from './Body';
import Label from './Label';
import Pagination from './Pagination';
import Text from './Text';

export default {
  BodyNodes,
  Body,
  Label,
  Pagination,
  Text
};
