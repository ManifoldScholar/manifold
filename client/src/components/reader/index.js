export Section from "./Section";
export Annotation from "./Annotation";
export Resource from "./Resource";
export AppearanceMenuBody from "./AppearanceMenuBody";
export AppearanceMenuButton from "./AppearanceMenuButton";
export Footer from "./Footer";
export FooterMenu from "./FooterMenu";
export Header from "./Header";
export ReturnMenu from "./ReturnMenu";
export StaticTextSample from "./StaticTextSample";
export TextTitles from "./TextTitles";
export Toc from "./Toc";
export TocDrawer from "./TocDrawer";
export MyAnnotations from "./MyAnnotations";
export VisibilityButton from "./VisibilityButton";
