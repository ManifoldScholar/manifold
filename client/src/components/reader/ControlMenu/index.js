import AppearanceMenuBody from "./AppearanceMenuBody";
import AppearanceMenuButton from "./AppearanceMenuButton";
import VisibilityButton from "./VisibilityButton";
import NotesButton from "./NotesButton";

export default {
  AppearanceMenuBody,
  AppearanceMenuButton,
  VisibilityButton,
  NotesButton
};
