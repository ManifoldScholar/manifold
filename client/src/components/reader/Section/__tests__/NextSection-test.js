import React from "react";
import renderer from "react-test-renderer";
import { wrapWithRouter } from "test/helpers/routing";
import { Section } from "components/reader";

describe("NextSection component", () => {
  const sectionId = "1234-5678-9000";
  const sectionsMap = [
    { id: "1234-5678-9000", name: "First Section" },
    { id: "2345-5678-1111", name: "Second Section" }
  ];
  const textId = "2345-5678-1111";
  const typography = {
    margins: [0, 1, 2]
  };

  it("renders correctly", () => {
    const component = renderer.create(
      wrapWithRouter(
        <Section.NextSection
          sectionId={sectionId}
          sectionsMap={sectionsMap}
          textId={textId}
          typography={typography}
        />
      )
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
