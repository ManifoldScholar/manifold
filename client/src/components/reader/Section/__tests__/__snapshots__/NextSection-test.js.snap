// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextSection component renders correctly 1`] = `
<section
  className="section-next-section"
>
  <div
    className="container-focus container-width-undefined"
  >
    <a
      href="/read/2345-5678-1111/section/2345-5678-1111"
      onClick={[Function]}
    >
      <header>
        Next Chapter
      </header>
      <h3>
        <div>
          Second Section
        </div>
      </h3>
    </a>
  </div>
</section>
`;
